addons/mail/static/src/core/message_service.js

```diff
@@ -70,10 +70,9 @@ export class MessageService {
      * @returns {number}
      */
     getLastMessageId() {
-        return Object.values(this.store.messages).reduce(
-            (lastMessageId, message) => Math.max(lastMessageId, message.id),
-            0
-        );
+        return Object.values(this.store.messages)
+            .filter((message) => Number.isInteger(message.id))
```

Comment:
Ignore other transient will make the `getNextTemporaryId` collapse all transient to the same id, effectively allowing only one.

The correct fix is probably to always use `getNextTemporaryId`, also for pending messages.